core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -282,7 +283,8 @@ memories_deinstantiate(WASMModuleInstance *module_inst)
         memory_deinstantiate(memory);
 #endif
 
-#if WASM_ENABLE_MULTI_MODULE == 0 && WASM_ENABLE_SHARED_MEMORY != 0
+#if WASM_ENABLE_MULTI_MODULE == 0
+#if WASM_ENABLE_SHARED_MEMORY != 0
         /* for spawned only */
         if (!shared_memory_is_shared(memory)) {
             continue;
```

Comment:
Yes. need to `wasm_runtime_free(memory)`. 

`continue` is still necessary because `shared_memory_get_reference()` works only when ` bh_assert(shared_memory_is_shared(memory));`.